[Style] 약 추가 두번째 UI 구현

seungchan2 opened the pull request:
## 🌴 PR 요약

<!-- PR의 내용을 요약해주세요. -->
일단 기본적인 UI만 구현을 했습니다 행님들
캘린더는 약 추가 UI 다 구현하면 건들게요 !
반복해서 쓰는 함수들을 프로토콜을 만들어서 빼놨어요 (ViewPresentable)

🌱 작업한 브랜치

- feature/#179

🌱 작업한 내용

- 기본 UI 구현
- 프로토콜화
- 여백 지우기 

## 📸 스크린샷
<img src = "https://user-images.githubusercontent.com/80672561/165947306-b356e02b-02c0-49e7-8036-7437998b9caa.png"  width = "250"/>


## 📮 관련 이슈

- Resolved: #179

seungchan2 committed:
[#176] PillDay 모달 구현

seungchan2 committed:
[#179] PopUpProtocol

seungchan2 committed:
[#179] TimeViewModel

seungchan2 committed:
[#179] View 완성

seungchan2 committed:
[Style] 약 추가 두번째 뷰 UI 구현 (#179)

### Description

- 기본 UI만 구현
- ViewPresentable 프로토콜 추가
- 캘린더는 추후에 작업할 예정

Resolve #179

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/Cell/PillDayTableViewCell.swift:
```diff
@@ -0,0 +1,59 @@
+        $0.image = Image.isVImage
+    }
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
```
이거 궁금한데 설명 점 ...

seungchan2 replied:
기본적으로 TableViewCell을 코드로 구현 할 때 초기화 목적으로 적어뒀습니다! (필수!)
https://ios-development.tistory.com/445 
종권님 블로그에 간단하게 설명되어 있는데 참고하셔도 좋을 거 같아용 !!!

seungchan2 replied:
zzzzz 아차차

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/View/PillDayView.swift:
```diff
@@ -0,0 +1,67 @@
+        }
+    }
+    
+    private func setupConstraints() {
```
`ViewPresentable` 프로토콜을 채택하면 얘들이 자동으로 들어가는데, 거기다가 `private` 제한 걸어줬다! 제가 이해한 게 맞나요 ?

seungchan2 replied:
이렇게 하면 안됨! 접근제어 사용 금지!! 굳이 사용하자면 internal 키워드인데 굳이 안해줘도 댐 !

seungchan2 replied:
이 부분은 제가 .. 실수를 한 거 같은데 생각해보니까 private하게 관리할 필요가 없겠군요 . . 
수정해서 반영할게요 !

seungchan2 replied:
굳이 사용하려면 internal 키워드를 사용할 수 있을텐데 그렇다면 굳이 접근 제어를 안 해도 될 거 같아요!

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/Cell/PillDayTableViewCell.swift:
```diff
@@ -0,0 +1,59 @@
+import SnapKit
+import Then
+
+final class PillDayTableViewCell: UITableViewCell {
```
프로토콜 채택해주기 ~

seungchan2 replied:
이건 보지를 못했네요 .. 나중에 이슈파서 한 번에 관리할게요 >.<

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/View/PillDayView.swift:
```diff
@@ -0,0 +1,67 @@
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
```
위에서 비가용성 정의 해줬으니까 여기도 해주면 더 좋을 듯한 ...

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/View/PillPeriodView.swift:
```diff
@@ -0,0 +1,64 @@
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
```
여기도 !

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/ViewController/AddPillFirstViewController.swift:
```diff
@@ -40,6 +42,20 @@ final class AddPillFirstViewController: BaseViewController {
         navigationController?.isNavigationBarHidden = true
     }
     
+    private func presentView() {
```
네이밍 좋네요 ... MVVM 굳

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/ViewController/AddPillFirstViewController.swift:
```diff
@@ -69,9 +88,9 @@ final class AddPillFirstViewController: BaseViewController {
                 case .everyday:
                     print("everyday")
                 case .day:
-                    print("day")
+                    self.presentView()
                 case .period:
-                  print("period")
+                    self.presentPeriodView()
```
나 이거 쇽샥 할래 ... 한 6개월 뒤에 ...

seungchan2 replied:
한 달 전에 공부한 내용이라 . . 안 까먹기위해 . . 공부해서 더 적용해보게씀다

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/ViewController/PillDayViewController.swift:
```diff
@@ -0,0 +1,89 @@
+
+final class PillDayViewController: BaseViewController {
+
+    let pillDayView = PillDayView()
+    let disposeBag = DisposeBag()
+    let weekObservable = Observable.of(["월", "화", "수", "목", "금", "토", "일"])
+    let dataObservable = Observable.of("")
+    private var pillTimeViewModel = PillTimeViewModel()
+    var delegate: PopUpDelegate?
```
질문2 ... 접근 제한자 왜 뷰모델에만 걸어주셨나요 ...?

seungchan2 replied:
일단 변명을 하자면 ㅋㅋ 이 코드는 한달전에 작성한 코드로 . . 제가 실수를 했네요. . delegate 앞 부분에 weak도 추가해놓을게요 ~

Taehyeon-Kim commented:
김승찬 그만 잘해 .

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/ViewController/PillDayViewController.swift:
```diff
@@ -0,0 +1,89 @@
+        super.style()
+        view.backgroundColor = UIColor(white: 0.1, alpha: 0.5)
+    }
+    
+    
+
+    
+
```
그 ... 제가 빈칸이나 주석 얘기는 잘 안 하는데 이 파일은 좀 정리를 해주셔야 할듯한 ...

seungchan2 replied:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ 아니 이미 머지한 줄 알았는데 . .죄송합니다

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillSecond/View/AddPillSecondView.swift:
```diff
@@ -0,0 +1,119 @@
+        setupConstraints()
+        setCalendarStyle()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
```
여기두 ...

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillSecond/View/AddPillSecondView.swift:
```diff
@@ -0,0 +1,119 @@
+    }
+    
+    private func setupView() {
+        [navigationView, pillQuestionLabel, pillPeriodLabel, calendar, checkBoxButton, startThreeMonthLabel, periodInformationLabel, nextButton].forEach {
+            addSubview($0)
```
맞당 이거 EasyKit에 `addSubViews` 함수가 있더라고요... 얼마 전에 뜯어보다가 알았슴

EunHee-Jeong commented on SobokSobok/SobokSobok/Presentation/Pill/PillSecond/ViewController/AddPillSecondViewController.swift:
```diff
@@ -0,0 +1,29 @@
+
+    }
+    
+    override func style() {
+        view.backgroundColor = .white
+        tabBarController?.tabBar.isHidden = true
+        navigationController?.isNavigationBarHidden = true
+    }
```
super 해주기!

EunHee-Jeong reviewed (commented):
열심히 읽었다 ...

seungchan2 committed:
Merge branch 'develop' into feature/#179

seungchan2 committed:
[#179] 코드 리뷰 반영

### Description

- 코드 리뷰 반영

Resolve #179

seungchan2 committed:
[#179] BaseView 작업

### Description

- ViewPresentable 프로토콜에서 BaseView를 하나 만들고 override해서 사용

Resolve #179

Taehyeon-Kim commented on SobokSobok/SobokSobok/Presentation/Pill/PillFirst/ViewController/PillPeriodViewController.swift:
```diff
@@ -0,0 +1,32 @@
+final class PillPeriodViewController: BaseViewController {
+
+    let pillPeriodView = PillPeriodView()
+    var delegate: PopUpDelegate?
```
memory leak 이슈 때문에 앞에는 weak 붙여주면 좋을 것 같네요 굿!

seungchan2 replied:
BaseView 작업할 때 수정해놨습니뎅 ~

Taehyeon-Kim reviewed (approved):
좋다.. 코드!

seungchan2 reviewed (commented):
ㅋ
